client/src/components/TinymceInlineToolbar/TinymceInlineToolbar.js

```diff
@@ -0,0 +1,243 @@
+    const buffer = 5;
+    const margin = 8;
+    const spaceNeeded = toolbarHeight + margin + buffer;
+    const mceToolbarBottom = this.mceToolbar ?
+      this.mceToolbar.getBoundingClientRect().bottom : 0;
+    const mceStatusbarTop = this.mceStatusbar ?
+      windowHeight - this.mceStatusbar.getBoundingClientRect().top : 0;
+    const blockedTop = Math.max(0, mceToolbarBottom, iframeRect.top);
+    const blockedBottom = Math.max(0, mceStatusbarTop, windowHeight - iframeRect.bottom);
+    const spaceTop = (selection.top + iframeRect.top) - blockedTop;
+    const spaceBottom = windowHeight - iframeRect.top - selection.bottom - blockedBottom;
+    const editorHeight = windowHeight - blockedTop - blockedBottom;
+    const topOffset = 10;
+    let className = '';
+    let top = 0;
+    let left = 0;
+
+    if (spaceTop >= editorHeight || spaceBottom >= editorHeight) {
+      return this.hide();
+    }
+
+    if (this.bottom) {
+      if (spaceBottom >= spaceNeeded) {
+        className = ' mce-arrow-up';
+        top = selection.bottom + iframeRect.top + scrollY + topOffset;
+      } else if (spaceTop >= spaceNeeded) {
+        className = ' mce-arrow-down';
+        top = (selection.top + iframeRect.top + scrollY) - toolbarHeight - margin;
+      }
+    } else if (spaceTop >= spaceNeeded) {
+      className = ' mce-arrow-down';
+      top = (selection.top + iframeRect.top + scrollY) - toolbarHeight - margin;
+    } else if (
+      spaceBottom >= spaceNeeded &&
+      editorHeight / 2 > (selection.bottom + iframeRect.top) - blockedTop
+    ) {
+      className = ' mce-arrow-up';
+      top = selection.bottom + iframeRect.top + scrollY + topOffset;
+    }
+
+    if (typeof top === 'undefined') {
+      top = scrollY + blockedTop + buffer;
+    }
+
+    left = (selectionMiddle - (toolbarWidth / 2)) + iframeRect.left + scrollX;
+
+    if (selection.left < 0 || selection.right > iframeRect.width) {
+      left = iframeRect.left + scrollX + ((iframeRect.width - toolbarWidth) / 2);
+    } else if (toolbarWidth >= windowWidth) {
+      className += ' mce-arrow-full';
+      left = 0;
+    } else if (
+      (left < 0 && selection.left + toolbarWidth > windowWidth) ||
+      (left + toolbarWidth > windowWidth && selection.right - toolbarWidth < 0)
+    ) {
+      left = (windowWidth - toolbarWidth) / 2;
+    } else if (left < iframeRect.left + scrollX) {
+      className += ' mce-arrow-left';
+      left = selection.left + iframeRect.left + scrollX;
+    } else if (left + toolbarWidth > iframeRect.width + iframeRect.left + scrollX) {
+      className += ' mce-arrow-right';
+      left = (selection.right - toolbarWidth) + iframeRect.left + scrollX;
+    }
+
+    toolbarEl.className = toolbarEl.className.replace(/ ?mce-arrow-[\w]+/g, '') + className;
+    this.setStyles({ left, top });
+
+    return this;
+  }
+}
+
+/**
+ * @param  tinymce selection object
+ * @param  {Boolean}
+ * @param  {Array} array of tag names - e.g. "A"
```

Comment:
almost right with the JSDocs :D
` * @param {Object} selection tinymce selection object`
where the first selection is the name of the param